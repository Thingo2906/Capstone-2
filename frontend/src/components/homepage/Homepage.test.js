import React from "react";
import Homepage from "./Homepage";
import { render} from "@testing-library/react";
import UserContext from "../../auth/UserContext";

describe("Homepage component", function(){
    test("render without crashing", function(){
        const currentUser = {};

        render(
          <UserContext.Provider value={{ currentUser }}>
            <Homepage />
          </UserContext.Provider>
        );
    });
    test("Matches snapshot", function(){
        const currentUser = {};
        const { asFragment } = render(
          <UserContext.Provider value={{ currentUser }}>
            <Homepage />
          </UserContext.Provider>
        );
        expect(asFragment()).toMatchSnapshot();
    });
    test("renders Homepage when user logged in", function(){
        const currentUser={ username: "testuser", firstName: "test", lastName: "user"};
        const { getByText } = render(
          <UserContext.Provider value={{ currentUser }}>
            <Homepage />
          </UserContext.Provider>
        );
        expect(getByText(`Welcome back! ${currentUser.firstName}`)).toBeInTheDocument();
    });
})