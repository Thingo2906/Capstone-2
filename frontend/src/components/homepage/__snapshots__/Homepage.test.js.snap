// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Homepage component Matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="Homepage"
  >
    <div
      class="text-center"
    >
      <h1
        class="mb-4 font-weight-bold"
      >
        MOVIES
      </h1>
      <p>
        All the movies in one. Enjoys your movie with us!.
      </p>
      <h2>
        Welcome back! 
      </h2>
    </div>
  </div>
</DocumentFragment>
`;
